reference-implementation/lib/writable-stream.js

```diff
@@ -769,39 +770,40 @@ function WritableStreamDefaultControllerGetDesiredSize(controller) {
   return controller._strategyHWM - controller._queueTotalSize;
 }
 
-function WritableStreamDefaultControllerUpdateBackpressureIfNeeded(controller, oldBackpressure) {
-  const stream = controller._controlledWritableStream;
-  if (stream._state !== 'writable') {
-    return;
-  }
-
-  const backpressure = WritableStreamDefaultControllerGetBackpressure(controller);
-  if (oldBackpressure !== backpressure) {
-    WritableStreamUpdateBackpressure(stream, backpressure);
-  }
-}
-
 function WritableStreamDefaultControllerWrite(controller, chunk, chunkSize) {
-  const stream = controller._controlledWritableStream;
-
-  assert(stream._state === 'writable');
-
   const writeRecord = { chunk };
 
-  const oldBackpressure = WritableStreamDefaultControllerGetBackpressure(controller);
-
   try {
     EnqueueValueWithSize(controller, writeRecord, chunkSize);
   } catch (enqueueE) {
     WritableStreamDefaultControllerErrorIfNeeded(controller, enqueueE);
     return;
   }
 
-  WritableStreamDefaultControllerUpdateBackpressureIfNeeded(controller, oldBackpressure);
+  const stream = controller._controlledWritableStream;
+  if (WritableStreamCloseQueuedOrInFlight(stream) === false) {
+    const backpressure = WritableStreamDefaultControllerGetBackpressure(controller);
+    WritableStreamUpdateBackpressure(stream, backpressure);
+  }
 
   WritableStreamDefaultControllerAdvanceQueueIfNeeded(controller);
 }
 
+function WritableStreamDefaultControllerStart(controller) {
+  const startResult = InvokeOrNoop(controller._underlyingSink, 'start', [controller]);
+  Promise.resolve(startResult).then(
+    () => {
+      controller._started = true;
+      // TODO: Test this.
```

Comment:
I no longer remember what I wanted to do. It seems this logic is well tested by start.js basically. I've removed this TODO.